src/lib/forms/FeedbackLabel.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from "react";
+import { Label, Icon } from "semantic-ui-react";
+import { flattenAndCategorizeErrors } from "../utils";
+import { InvenioPopup } from "../elements/accessibility";
+import PropTypes from "prop-types";
+
+export class FeedbackLabel extends Component {
+  constructor(props) {
+    super(props);
+
+    const { errorMessage } = props;
+    const { flattenedErrors = {}, severityChecks = {} } =
+      flattenAndCategorizeErrors(errorMessage);
+
+    // Get the first error and severity data, defaulting to empty values if not present
+    const errorText = Object.values(flattenedErrors)[0] || "";
+    const severityData = Object.values(severityChecks)[0] || {};
+
+    // Destructure severityData, ensuring default values for missing keys
+    const severityLevel = severityData?.severity || "";
+    const severityMessage = severityData?.message || "";
+    const severityDescription = severityData?.rule_info || "";
+
+    this.state = {
+      errorText,
+      severityInfo: { severityLevel, severityMessage, severityDescription },
+      prompt: !severityLevel && !!errorText,
+    };
+  }
+
+  render() {
+    const { errorText, severityInfo, prompt } = this.state;
+
+    // Return null if neither errorText nor severityMessage exists
+    if (!errorText && !severityInfo.severityMessage) {
+      return null;
+    }
+
+    return (
+      <Label pointing="left" className={severityInfo.severityLevel} prompt={prompt}>
+        {/* Display severity message with a popup if it exists */}
+        {severityInfo.severityMessage && severityInfo.severityDescription && (
```

Comment:
nit, for readability:
```suggestion
    const hasError = severityInfo.severityMessage && severityInfo.severityDescription;
    return (
      <Label pointing="left" className={severityInfo.severityLevel} prompt={prompt}>
        {/* Display severity message with a popup if it exists */}
        { hasError && (
```